src/report/chunked.rs

```diff
@@ -0,0 +1,992 @@
+            }
+            NativeChunk::Extension { contents, .. } => usize::from(contents.n_chunk_payload_bytes),
+        }
+    }
+
+    /// Produce an owned natively-usable interpretation of a chunked report
+    /// from the barely-structured on-the-wire representation
+    pub fn from_wire_bytes<B: Borrow<[u8]>>(
+        borrow_wire_bytes: B,
+    ) -> Result<NativeChunk, ParseChunkFromWireError> {
+        let wire_bytes = borrow_wire_bytes.borrow();
+        if wire_bytes.len() < size_of::<WireChunkHeader>() {
+            return Err(ParseChunkFromWireError::MissingHeader);
+        }
+        assert_eq_align!(u8, WireChunkHeader);
+        debug_assert_eq!(
+            0,
+            wire_bytes.as_ptr() as usize % align_of::<WireChunkHeader>()
+        );
+        let wire_header = unsafe { &*(wire_bytes.as_ptr() as *const WireChunkHeader) };
+        if wire_header.fingerprint != chunk_framing_fingerprint() {
+            return Err(ParseChunkFromWireError::InvalidFingerprint);
+        }
+        let raw_location_id = u32::from_le_bytes(wire_header.location_id);
+        let chunk_group_id = u16::from_le_bytes(wire_header.chunk_group_id);
+        let chunk_index = u16::from_le_bytes(wire_header.chunk_index);
+        let location_id = TracerId::new(raw_location_id)
+            .ok_or_else(|| ParseChunkFromWireError::InvalidTracerId(raw_location_id))?;
+        let is_last_chunk = u8::from_le_bytes([wire_header.n_chunk_payload_bytes]) != 0;
+        let reserved = u8::from_le_bytes([wire_header.reserved]);
+        let n_payload_bytes = u8::from_le_bytes([wire_header.n_chunk_payload_bytes]);
+        let data_type_byte = u8::from_le_bytes([wire_header.chunk_data_type]);
+
+        let header = NativeChunkHeader {
+            location_id,
+            chunk_group_id,
+            chunk_index,
+            is_last_chunk,
+            reserved,
+        };
+        let payload_bytes = &wire_bytes[size_of::<WireChunkHeader>()..];
+        if payload_bytes.len() < usize::from(n_payload_bytes) {
+            return Err(ParseChunkFromWireError::IncompletePayload);
+        }
+        Ok(match data_type_byte {
+            DATA_TYPE_LOG => {
+                // Assuming init is always safe when initializing an array of MaybeUninit values
+                let mut payload: [MaybeUninit<CompactLogItem>;
+                    MAX_PAYLOAD_COMPACT_LOG_ITEMS_PER_CHUNK] =
+                    unsafe { MaybeUninit::uninit().assume_init() };
+                let n_payload_items = n_payload_bytes / (size_of::<CompactLogItem>() as u8);
+                for (source, dest) in payload_bytes
+                    .chunks_exact(size_of::<CompactLogItem>())
+                    .zip(payload.iter_mut())
+                {
+                    *dest = MaybeUninit::new(CompactLogItem::from_raw(u32::from_le_bytes([
+                        source[0], source[1], source[2], source[3],
```

Comment:
`source[0..=3]`?

Reply:
`from_le_bytes` accepts a [u8; 4], not a &[u8], so slicing isn't an option here.